sklearn/ensemble/gradient_boosting.py

```diff
@@ -1866,3 +1872,310 @@ def apply(self, X):
+        """
+        if isinstance(self.cv_n_estimators, (numbers.Integral, np.integer)):
+            print('heee')
+            cv_n_estimators = np.array([self.cv_n_estimators, ], dtype=np.int)
+        else:
+            cv_n_estimators = np.array(self.cv_n_estimators, dtype=np.int,
+                                       copy=False)
+            if cv_n_estimators.ndim != 1:
+                raise ValueError("The cv_n_estimators is expected to be a 1D "
+                                 "array of possible n_estimators values. %r "
+                                 "is not" % cv_n_estimators)
+
+        cv_n_estimators.sort()
+        cv = check_cv(self.cv)
+        scorer = check_scoring(self, scoring=self.scoring)
+
+        # We will be parallelizing across the splits
+        # The same base estimator must be used for each split for all
+        # boost iterations
+        base_estimator = GradientBoostingClassifier(
+            n_estimators=cv_n_estimators[0], warm_start=True,
+            random_state=self.random_state)
+
+        parallel = Parallel(n_jobs=self.n_jobs, pre_dispatch=self.pre_dispatch)
+        out = parallel(delayed(_fit_score_all_stages)(clone(base_estimator),
+                                                      X, y, sample_weight,
+                                                      train, test, scorer,
+                                                      cv_n_estimators)
+                       for train, test in cv.split(X, y, groups=sample_groups))
+
+        # Store the mean cross-val scores across all splits for each stage
+        self.scores_ = np.asarray(out).mean(axis=0)
+        # Store the sorted cv_n_estimators
+        self.cv_n_estimators_ = cv_n_estimators
+        self.n_estimators_ = cv_n_estimators[self.scores_.argmax()]
+
+        # Set the final n_estimators based on the best score and do a full fit
+        # set_params wont work as n_estimators is not a param of GBCV
+        self.n_estimators = self.n_estimators_
+        super(GradientBoostingClassifierCV, self).fit(X, y, sample_weight)
+        return self
+
+
+def _fit_score_all_stages(estimator, X, y, sample_weight, train, test,
+                          scorer, cv_n_estimators):
+    """Fit all stages and compute scores after each stage for given cv split"""
+    X_train, y_train = X[train], y[train]
+    X_test, y_test = X[test], y[test]
+
+    if sample_weight is not None:
+        weight_train, weight_test = sample_weight[train], sample_weight[test]
+    else:
+        weight_train = weight_test = None
+
+    all_stage_scores = np.zeros(cv_n_estimators.shape, dtype=np.float64)
+
+    for i, n_estimators in enumerate(cv_n_estimators):
+        estimator.set_params(n_estimators=n_estimators)
+        estimator.fit(X_train, y_train, sample_weight=weight_train)
+        all_stage_scores[i] = scorer(estimator, X_test, y_test,
+                                     sample_weight=weight_test)
```

Comment:
But eventually we would want to support `sample_weights` at `GridSearchCV` no? I'll push a commit which documents this... Let me know if you still want it removed... (Moreover the `GradientBoostingClassifier.fit` supports sw, and users might expect a similar interface maybe?)